viewer/packages/360-images/src/icons/IconCollection.ts

```diff
@@ -0,0 +1,211 @@
+    const iconsSprites = new InstancedIconSprite(
+      points.length,
+      sharedTexture,
+      this.MIN_PIXEL_SIZE,
+      this.MAX_PIXEL_SIZE,
+      0.5
+    );
+    iconsSprites.setPoints(points);
+
+    this._hoverIconTexture = this.createHoverIconTexture();
+    this._sharedTexture = sharedTexture;
+    this._icons = this.initializeImage360Icons(points, sceneHandler, onBeforeSceneRendered);
+
+    const octreeBounds = IconOctree.getMinimalOctreeBoundsFromIcons(this._icons);
+    const octree = new IconOctree(this._icons, octreeBounds, 4);
+
+    const clusterSprites = new InstancedIconSprite(
+      points.length,
+      this.createClusterTexture(),
+      this.MIN_PIXEL_SIZE * 2,
+      this.MAX_PIXEL_SIZE,
+      2.0
+    );
+
+    const projection = new Matrix4();
+    onBeforeSceneRendered.subscribe(({ camera }) => {
+      const a = octree.getLODByScreenArea(
+        0.005,
+        projection.copy(camera.projectionMatrix).multiply(camera.matrixWorldInverse)
+      );
+
+      const nodes = [...a];
+      const clusterPoints = nodes.filter(p => p.data === null).map(p => octree.getPointCenterOfNode(p)!);
+      clusterSprites.setPoints(clusterPoints);
+      const leafData = nodes.filter(p => p.data !== null).flatMap(p => p.data.data);
+      this._icons.forEach(p => (p.visible = false));
+      leafData.forEach(p => (p.visible = true));
+      iconsSprites.setPoints(leafData.map(p => p.position));
+    });
+
+    this._sceneHandler = sceneHandler;
+    this._iconsSprite = iconsSprites;
+    this._clusterSprites = clusterSprites;
+
+    sceneHandler.addCustomObject(iconsSprites);
+    sceneHandler.addCustomObject(clusterSprites);
+  }
+
+  private initializeImage360Icons(
+    points: Vector3[],
+    sceneHandler: SceneHandler,
+    onBeforeSceneRendered: EventTrigger<BeforeSceneRenderedDelegate>
+  ): Image360Icon[] {
+    return points.map(
+      point =>
+        new Image360Icon(
+          point,
+          this._hoverIconTexture,
+          sceneHandler,
+          this.MIN_PIXEL_SIZE,
+          this.MAX_PIXEL_SIZE,
+          onBeforeSceneRendered
+        )
+    );
+  }
+
+  public dispose(): void {
+    this._sceneHandler.removeCustomObject(this._clusterSprites);
+    this._sceneHandler.removeCustomObject(this._iconsSprite);
+    this._clusterSprites.dispose();
+    this._iconsSprite.dispose();
+    this._onRenderTrigger.unsubscribeAll();
+    this._sharedTexture.dispose();
+  }
+
+  private createClusterTexture(): CanvasTexture {
+    const canvas = document.createElement('canvas');
+    const textureSize = this.MAX_PIXEL_SIZE;
+    canvas.width = textureSize;
+    canvas.height = textureSize;
+
+    const halfTextureSize = textureSize / 2;
+
+    const context = canvas.getContext('2d')!;
+    drawClusterRings();
+    drawInnerCircle();
+    drawOuterCircle();
+
+    return new CanvasTexture(canvas);
+
+    function drawClusterRings() {
+      context.beginPath();
+      context.lineWidth = textureSize / 22;
```

Comment:
As `22` is used as part of assigning few `lineWidth` value, can this be a `const` variable like `MAX_PIXEL_SIZE/MIN_PIXEL_SIZE`?